### PR TITLE
feat(client): create shipping UI

### PR DESCRIPTION
**Pull request by DavidLiang01**
#

## Changes this PR introduces

- UI for Map
- UI for Shipping Table

### You can use the following link to test the PR: https://deploy-preview-60--soen-390-team-07.netlify.app
- When hover on table rows, it should show related route 
- Display map 
- Able to toggle button in table 
- Map doesn't show orders in "delivered" and "cancelled" status

## ClickUp reference

CU-n32e3j

## Notes for reviewers
- The displayed data for shipping table are mocked

### If testing locally:
- Require  NEXT_PUBLIC_MAPBOX key in .env.local
- Look on Discord for apiKey value

## Checklist

- [x] Have I renamed the title of this pull request to follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)?
- [ ] I will squash and merge when my PR is approved.
- [x] I have added tests to cover my changes (if applicable).

### CODE REVIEW

File: client/src/components/ShippingTable.jsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        );
+
+        // TODO: Update new changes to DB
+
+        return shpmtArr;
+      });
+    }
+  };
+
+  const handleHover = (e, eType) => {
```
Comment (gordonpn) [inline client/src/components/ShippingTable.jsx:65]: Hovering with shipment lines status is a really nice touch!

File: client/src/components/ShippingTable.jsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                    onMouseOver={(e) => {
+                      handleHover(e, 'over');
+                    }}
+                    onMouseOut={(e) => {
+                      handleHover(e, 'out');
+                    }}
+                  >
+                    <StyledTableCell>{shipment._id}</StyledTableCell>
+                    <StyledTableCell>
+                      <Button id={shipment._id} onClick={handleShpmtStateChange}>
```
Comment (gordonpn) [inline client/src/components/ShippingTable.jsx]: Could you add a tooltip or something to show what the next status would be? (Like in ClickUp)

image

File: client/src/stores/ShippingStore.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { makeObservable, observable, action, computed } from 'mobx';
+
+class ShippingStore {
```
Comment (gordonpn) [inline client/src/stores/ShippingStore.js:3]: Will the `ShippingStore` still be necessary when the shipping API is available?

**Review by gordonpn — COMMENTED** [review]
The map is a really nice looking feature like the others said!

File: client/src/components/ShippingTable.jsx (hunk 2 above)
Comment (DavidLiang01) [reply client/src/components/ShippingTable.jsx]: I will make the requested change

File: client/src/stores/ShippingStore.js (hunk 3 above)
Comment (DavidLiang01) [reply client/src/stores/ShippingStore.js:3]: I was wondering that we shouldn't call the API every time we render components in the page. In addition, is it better if we call the APIs when the user logs in and initialize all the stores?

Comment (gordonpn) [reply client/src/stores/ShippingStore.js:3]: > I was wondering that we shouldn't call the API every time we render components in the page. In addition, is it better if we call the APIs when the user logs in and initialize all the stores?

Those two concerns are solved by `react-query` which we are already using in the inventory and admin pages.

Comment (DavidLiang01) [reply client/src/stores/ShippingStore.js:3]: I will look for more in details on the `shippingStore ` and `react-query`

Comment (DavidLiang01) [reply client/src/stores/ShippingStore.js:3]: @gordonpn  I will refactor once I have created the shipping APIs

Comment (gordonpn) [reply client/src/stores/ShippingStore.js:3]: @DavidLiang01 
> I will refactor once I have created the shipping APIs

Sounds good

**Review by gordonpn — DISMISSED** [review]

**Comment by DavidLiang01** [comment]
Task link: [CU-n32e3j Shipping page](https://app.clickup.com/t/n32e3j)

**Review by gordonpn — APPROVED** [review]

**Comment by Mackbellemore** [comment]
Task linked: [CU-n32e3j Shipping page](https://app.clickup.com/t/n32e3j)

**Review by Mackbellemore — APPROVED** [review]

**Pull request merged**